yarn-project/rollup-provider/src/app.ts

```diff
@@ -0,0 +1,194 @@
+    const output: { [key: string]: string } = {};
+    for (const [key, value] of Object.entries(roots)) {
+      output[key] = value.toString();
+    }
+    ctx.body = {
+      roots: output,
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/get-unverified', async (ctx: Koa.Context) => {
+    const from = +ctx.query.from!;
+    const take = +ctx.query.take!;
+    const blocks = await node.getUnverifiedData(from, take);
+    const strs = blocks.map(x => x.toBuffer().toString('hex'));
+    ctx.set('content-type', 'application/json');
+    ctx.body = {
+      unverified: strs,
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/get-pending-tx', async (ctx: Koa.Context) => {
+    const hash = ctx.query.hash!;
+    const txHash = new TxHash(Buffer.from(hash as string, 'hex'));
+    const tx = await node.getPendingTxByHash(txHash);
+    ctx.set('content-type', 'application/json');
+    ctx.body = {
+      tx: tx == undefined ? undefined : txToJson(tx),
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/contract-index', async (ctx: Koa.Context) => {
+    const leaf = ctx.query.leaf!;
+    const index = await node.findContractIndex(Buffer.from(leaf as string, 'hex'));
+    ctx.set('content-type', 'application/json');
+    ctx.body = {
+      index,
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/contract-path', async (ctx: Koa.Context) => {
+    const leaf = ctx.query.leaf!;
+    const path = await node.getContractPath(BigInt(leaf as string));
+    ctx.set('content-type', 'application/json');
+    ctx.body = {
+      path: path.toString(),
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/data-path', async (ctx: Koa.Context) => {
+    const leaf = ctx.query.leaf!;
+    const index = BigInt(leaf as string);
+    const path = await node.getDataTreePath(index);
+    ctx.set('content-type', 'application/json');
+    const pathAsString = path.toString();
+    ctx.body = {
+      path: pathAsString,
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/l1-l2-path', async (ctx: Koa.Context) => {
+    const leaf = ctx.query.leaf!;
+    const path = await node.getL1ToL2MessagesTreePath(BigInt(leaf as string));
+    ctx.set('content-type', 'application/json');
+    ctx.body = {
+      path: path.toString(),
+    };
+    ctx.status = 200;
+  });
+
+  router.get('/storage-at', async (ctx: Koa.Context) => {
+    logger('storage-at');
+    const address = ctx.query.address!;
+    const slot = ctx.query.slot!;
+    const value = await node.getStorageAt(AztecAddress.fromString(address as string), BigInt(slot as string));
+    ctx.set('content-type', 'application/json');
+    ctx.body = {
+      value: value?.toString('hex'),
+    };
+    ctx.status = 200;
+  });
+
+  router.post('/tx', checkReady, async (ctx: Koa.Context) => {
+    const stream = new PromiseReadable(ctx.req);
+    const postData = JSON.parse((await stream.readAll()) as string);
+    const tx = txFromJson(postData);
```

Comment:
We'll need to put in some validation here, to make sure we're not passing in garbage within a tx